src/mpyl/steps/deploy/k8s/chart.py

```diff
@@ -510,16 +517,40 @@ def to_white_list(
             for idx, host in enumerate(hosts if hosts else default_hosts)
         ]
 
-    def to_ingress(self) -> V1AlphaIngressRoute | None:
-        return (
-            V1AlphaIngressRoute.from_spec(
-                metadata=self._to_object_meta(name="ingress-routes"),
-                spec=self.deployment.traefik.ingress_routes.get_value(self.target),
-            )
+    def _replace_placeholders(self, traefik_object: dict | list):
+        for placeholder, replacement in {
+            PR_NUMBER_PLACEHOLDER: str(
+                self.step_input.run_properties.versioning.pr_number
+            ),
+            SERVICE_NAME_PLACEHOLDER: self.release_name,
+            NAMESPACE_PLACEHOLDER: self.namespace,
+            CLUSTER_ENV_PLACEHOLDER: get_cluster_config_for_project(
+                self.step_input.run_properties, self.project
+            ).cluster_env,
+        }.items():
+            traefik_object = replace_item(traefik_object, placeholder, replacement)
+
+        return traefik_object
+
+    def to_ingress(self) -> Optional[V1AlphaIngressRoute]:
+        """Converts the deployment traefik ingress routes configuration to a V1AlphaIngressRoute object."""
+        ingress_route_spec = (
+            self.deployment.traefik.ingress_routes.get_value(self.target)
             if self.deployment.traefik and self.deployment.traefik.ingress_routes
             else None
         )
 
+        if not ingress_route_spec:
+            return None
+
+        if self.target == Target.PULL_REQUEST:
```

Comment:
Why only for `PULL_REQUEST` ? Placeholders like `{SERVICE-NAME}` can be used in any environment, right?

Reply:
True, because I first made this only for the PR_NUMBER placeholder 😬